File: 프로그래머스/Level 2/대충 만든 자판.py
# https://school.programmers.co.kr/learn/courses/30/lessons/160586
def makeDict(keymap):
    d = {}
    
    for keym in keymap:
        for i in range(len(keym)):
            # 딕셔너리에 없다면
            if keym[i] not in d:
                d[keym[i]] = i + 1
            else: # 비교 -> 최소를 해야하니
                d[keym[i]] = min(i + 1, d[keym[i]])
    
    return d

def solution(keymap, targets):
    answer = []
    d = makeDict(keymap) # 딕셔너리 만들기
    
    # 타겟 순회
    for target in targets:
        totCnt = 0 # 총 키 누른 횟수
        for t in target:
            if t in d: # 딕셔너리에 있면
                totCnt += d[t]
            else:
                totCnt = -1
                break # 단, 목표 문자열을 작성할 수 없을 때는 -1을 저장
        
  
        answer.append(totCnt)
            
    return answer

File: 프로그래머스/Level 2/대충 만든 자판.swift
// https://school.programmers.co.kr/learn/courses/30/lessons/160586?language=swift
import Foundation

func solution(_ keymap:[String], _ targets:[String]) -> [Int] {
    var answer:[Int] = []
    
    // 딕셔너리 생성
    var dic:[String:Int] = [:]
    
    for i in 0..<keymap.count {
        let tmp = keymap[i].map { String($0) }
        //print(tmp)
        for j in 0..<tmp.count {
            if let key = dic[tmp[j]] {
                dic[tmp[j]]! = min(j+1, dic[tmp[j]]!)
            } else {
                dic[tmp[j]] = j+1
            }
        }
    }
    
    for t in targets {
        let target = t.map{ String($0) }
        var cnt:Int = 0
        
        for tar in target {
            if let val = dic[tar] {
                cnt += val
            } else {
                cnt = -1
                break
            }
        }
        
        answer.append(cnt)
    }
    
    return answer
}